utils.go

```diff
@@ -63,18 +63,115 @@ func IsDotFile(path string) bool {
 	return strings.HasPrefix(base, ".") && base != "."
 }
 
-var isVendorRegExp *regexp.Regexp
+// func IsVendor(filename string) bool {
+// 	for _, matcher := range data.VendorMatchers {
+// 		if matcher.MatchString(filename) {
+// 			return true
+// 		}
+// 	}
+// 	return false
+// }
+
+var allVendorRegExp regex.EnryRegexp
 
 // IsVendor returns whether or not path is a vendor path.
 func IsVendor(path string) bool {
-	return isVendorRegExp.MatchString(path)
+	return allVendorRegExp.MatchString(path)
+}
+
+func init() {
+	// We now collate all regexps from VendorMatchers to a single large regexp
+	// which is at least twice as fast to test than simply iterating & matching.
+	//
+	// ---
+	//
+	// We could test each matcher from VendorMatchers in turn i.e.
+	//
+	//  	func IsVendor(filename string) bool {
+	// 			for _, matcher := range data.VendorMatchers {
+	// 				if matcher.MatchString(filename) {
+	//					return true
+	//				}
+	//			}
+	//			return false
+	//		}
+	//
+	// Or naïvely concatentate all these regexps using groups i.e.
```

Comment:
```suggestion
	// Or natively concatentate all these regexps using groups i.e.
```

Reply:
I believe it is well-meant to be [naïvely](https://en.wiktionary.org/wiki/na%C3%AFvely).

Reply:
Maybe would be better to use `naively` then?